src/site/pages/digital/src/containers/ExprToCircuitPopup/generate.ts

```diff
@@ -1,73 +1,140 @@
+import {SetNameAction}           from "core/actions/SetNameAction";
+import {TranslateAction}         from "core/actions/transform/TranslateAction";
+import {CreateDeleteGroupAction} from "core/actions/deletion/DeleteGroupActionFactory";
+import {CreateDeselectAllAction,
+        SelectAction,
+        CreateGroupSelectAction} from "core/actions/selection/SelectAction";
 
 import {OrganizeMinDepth} from "core/utils/ComponentOrganizers";
 
-import {CreateICDataAction} from "digital/actions/CreateICDataAction";
-import {CreateNegatedGatesAction} from "digital/actions/simplification/NegatedGatesActionFactory";
+import {CreateICDataAction}    from "digital/actions/CreateICDataAction";
+import {FrequencyChangeAction} from "digital/actions/FrequencyChangeAction";
+import {InputPortChangeAction} from "digital/actions/ports/InputPortChangeAction";
+
+import {DigitalComponent, DigitalObjectSet} from "digital/models";
+import {ICData, IC, Clock, Label}           from "digital/models/ioobjects";
 
-import {DigitalComponent} from "digital/models";
-import {LED, ICData, IC} from "digital/models/ioobjects";
-import {DigitalCircuitInfo} from "digital/utils/DigitalCircuitInfo";
+import {DigitalCircuitInfo}  from "digital/utils/DigitalCircuitInfo";
 import {ExpressionToCircuit} from "digital/utils/ExpressionParser";
-import {GenerateTokens} from "digital/utils/ExpressionParser/GenerateTokens";
-import {OperatorFormat} from "digital/utils/ExpressionParser/Constants/DataStructures";
-import {CreateDeleteGroupAction} from "core/actions/deletion/DeleteGroupActionFactory";
+import {GenerateTokens}      from "digital/utils/ExpressionParser/GenerateTokens";
 
 
+export type ExprToCirGeneratorOptions = {
+    input?: InputTypes,
+    output?: OutputTypes,
+    isIC?: boolean,
+    connectClocksToOscope?: boolean,
+    label?: boolean,
+    format?: OperatorFormatLabel,
+    ops?: OperatorFormat,
+}
+
+export type InputTypes = "Button" | "Clock" | "Switch";
+export type OutputTypes = "Oscilloscope" | "LED";
+
+// TODO: Refactor this to a GroupAction factory once there is a better (and Action) algorithm to arrange the circuit
 export function Generate(info: DigitalCircuitInfo, expression: string,
-                         isIC: boolean, input: string, format: string,
-                         ops: OperatorFormat) {
-    // Set the operator format
-    if (format !== "custom")
-        ops = Formats.find(form => form.icon === format);
+                         options: ExprToCirGeneratorOptions) {
+    const input = options.input ?? "Switch";
+    const output = options.output ?? "LED";
+    const isIC = (output !== "Oscilloscope") ? (options.isIC ?? false) : false;
+    const connectClocksToOscope = options.connectClocksToOscope ?? false;
+    const label = options.label ?? false;
+    const format = options.format ?? "|";
+    const ops = (format === "custom") ? (options.ops ?? Formats[0]) : Formats.find(form => form.icon === format);
```

Comment:
you can define a `defaultOptions` constant above and then just have `const options = {...defaultOptions, ...userOptions}` and then also make `type ExprToCirGeneratorOptions` not have optional parameters and just make the parameter `userOptions: Partial<ExprToCirGeneratorOptions>`

Also if you want all the options to just be accessible by name without like `options.input` you can do
```ts
const {input, output, isIC, ...} = {...defaultOptions, ...userOptions};
```
but up to you, I think I would just go for using `options.whatever` tho